text/0000-inclusive-ranges.md

```diff
@@ -0,0 +1,79 @@
+- Feature Name: inclusive_range_syntax
+- Start Date: 2015-07-07
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Allow a `x...y` expression to create an inclusive range.
+
+# Motivation
+
+There are several use-cases for inclusive ranges, that semantically
+include both end-points. For example, iterating from `0_u8` up to and
+including some number `n` can be done via `for _ in 0..n + 1` at the
+moment, but this will fail if `n` is `255`. Furthermore, some iterable
+things only have a successor operation that is sometimes sensible,
+e.g., `'a'..'{'` is equivalent to the inclusive range `'a'...'z'`:
+there's absolutely no reason that `{` is after `z` other than a quirk
+of the representation.
+
+The `...` syntax mirrors the current `..` used for exclusive ranges:
+more dots means more elements.
+
+# Detailed design
+
+`std::ops` defines
+
+```rust
+pub struct RangeInclusive<T> {
+    pub start: T,
+    pub end: T,
+}
+```
+
+Writing `a...b` in an expression desugars to `std::ops::RangeInclusive
+{ start: a, end: b }`.
+
+This struct implements the standard traits (`Clone`, `Debug` etc.),
+but, unlike the other `Range*` types, does not implement `Iterator`
+directly, since it cannot do so correctly without more internal
```

Comment:
@gsingh93 The problem is that `0...u8::MAX` has 257 states: Some(i) for all i; and None. 
So you can't modify only one bound and get all the states. You have two options:

* extra flag for "inclusive case yielded"
* do a checked_add on each step, and if it fails then decrement the maximum instead of increment the minimum (symmetric logic applies for `next_back`)